src/Microsoft.DotNet.Internal.Health/AzureTableHealthReportProvider.cs

```diff
@@ -89,87 +71,44 @@ private static (string instance, string subStatus) ParseRowKey(string rowKey)
 
     public async Task UpdateStatusAsync(string serviceName, string instance, string subStatusName, HealthStatus status, string message)
     {
-        if (!TryGetUrlParts(out string baseUrl, out string sasQuery))
-            return;
-
         string partitionKey = EscapeKeyField(serviceName);
         string rowKey = GetRowKey(instance, subStatusName);
 
-        async Task Attempt()
-        {
-            HttpContent content = new ByteArrayContent(JsonSerializer.SerializeToUtf8Bytes(
-                new Entity {Status = status, Message = message},
-                s_jsonSerializerOptions
-            ));
-            content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");
-
-            string requestUri =
-                $"{baseUrl}(PartitionKey='{Uri.EscapeDataString(partitionKey)}',RowKey='{Uri.EscapeDataString(rowKey)}'){sasQuery}";
-            using HttpResponseMessage response = await _client.PutAsync(requestUri, content).ConfigureAwait(false);
-            response.EnsureSuccessStatusCode();
-        }
-
         try
         {
-            await _retry.RetryAsync(
-                Attempt,
-                e => _logger.LogWarning("Failed to update status for {service}/{subStatus}, retrying",
-                    serviceName,
-                    subStatusName),
-                e => true
-            ).ConfigureAwait(false);
+            await _tableClient.UpsertEntityAsync(new HealthReportTableEntity
+            {
+                PartitionKey = partitionKey,
+                RowKey = rowKey,
+                Status = status,
+                Message = message
+            });
         }
         catch (Exception e)
         {
             // Crashing out a service trying to report health isn't useful, log that we failed and move on
-            _logger.LogError(e, "Unable to update health status for {service}/{subStatus}", serviceName, subStatusName);
+            _logger.LogError(e, "Unable to update health status for {service}/{instance}|{subStatus}", serviceName, instance, subStatusName);
         }
     }
 
-    public Task<IList<HealthReport>> GetAllStatusAsync(string serviceName)
+    public async Task<IList<HealthReport>> GetAllStatusAsync(string serviceName)
     {
-        if (!TryGetUrlParts(out string baseUrl, out string sasQuery))
-            return Task.FromResult<IList<HealthReport>>(Array.Empty<HealthReport>());
-
         string partitionKey = EscapeKeyField(serviceName);
-        string filter = $"PartitionKey eq '{partitionKey}'";
 
-        async Task<IList<HealthReport>> Attempt()
+        AsyncPageable<HealthReportTableEntity> tableEntities = _tableClient.QueryAsync<HealthReportTableEntity>(x => x.PartitionKey == partitionKey);
```

Comment:
in this case we don't know if is going to return all the results, how many records do we have?, in this case I think will return1000 records per page but you are not iterating in the other pages, you need to go trough all the pages, like

List<HealthReportTableEntity> allEntites = new();
await foreach (Page<T> page in tableEntities.AsPages())
{
    allEntites.AddRange(page.Values);
}

so you can get all the results

Reply:
thanks for that! I should have read the documentation :D